R/utils-target-metadata.R

```diff
@@ -0,0 +1,65 @@
+#' Get Unique Target Task ID
+#'
+#' Retrieves the unique target task ID by extracting all target metadata
+#' and extracting the names of their `target_keys`. For valid config files
+#' these should be the same across all rounds and model tasks.
+#'
+#' @param config_tasks a list representation of the `tasks.json` config file.
+#'
+#' @return A character vector of unique target task ID names. Post v5.0.0
+#' this should be a single task ID name.
+#' @export
+get_target_task_id <- function(config_tasks) {
+  get_target_metadata(config_tasks) |>
+    purrr::map_chr(~ names(.x[["target_keys"]])) |>
+    unique()
```

Comment:
This allows hubs with implicit targets to be included.

```suggestion
  safe_names <- function(x) names(x) %||% ""
  res <- get_target_metadata(config_tasks) |>
    purrr::map_chr(~ safe_names(.x[["target_keys"]])) |>
    unique()
  if (identical(res, "")) NULL else res
```

Reply:
Thanks for catching this very obvious omision! I've added a very slightly modified version for this and also a bunch of tests too.